src/blockdata/script.rs

```diff
@@ -282,6 +282,25 @@ impl Script {
             self.0[34] == opcodes::all::OP_CHECKSIG.into_u8())
     }
 
+    /// Checks whether a script pubkey is a Segregated Witness (segwit) program.
+    #[inline]
+    pub fn is_witness_program(&self) -> bool {
+        // A scriptPubKey (or redeemScript as defined in BIP16/P2SH) that consists of a 1-byte
+        // push opcode (for 0 to 16) followed by a data push between 2 and 40 bytes gets a new
+        // special meaning. The value of the first push is called the "version byte". The following
+        // byte vector pushed is called the "witness program".
+        self.0.len() >= 4
+            && self.0.len() <= 42
+            // First byte version 0-16
+            && self.0[0] >= opcodes::all::OP_PUSHBYTES_0.into_u8()
+            && self.0[0] <= opcodes::all::OP_PUSHBYTES_16.into_u8()
```

Comment:
The first byte is required to be `PUSHBYTES_0` or on of the numeric opcodes (except negative 1).

Reply:
strange, you're commenting on an old commit

Reply:
I just noticed [this comment](https://github.com/rust-bitcoin/rust-bitcoin/pull/255#discussion_r307840597) on the new version of this and I addressed that, force-pushing now.

Reply:
Did a normal push instead of a force push. I can squash once I get ACKs if wanted.